### PR TITLE
Better document run_test.r

### PR DESCRIPTION
**Pull request by jdblischak**
### Description

This is a documentation-only PR to better advertise the existence of `run_test.r`. Most of the documentation only refers to `.bat`, `.sh`, and `.py`, with the occasional reference to `.pl`. Looking through the entire conda-forge org, I found [zero instances of `run_test.r`](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Frun_test.r&type=code); whereas, I found [329 instances of the better documented `run_test.py`](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Frun_test.py&type=code).

I had to do a lot of digging to convince myself that it should work:

* PR to fix `run_test.r` https://github.com/conda/conda-build/pull/1869
* Looking through the source code:
  * It has to be lowercase `.r` (R scripts are conventionally `.R`)
    https://github.com/conda/conda-build/blob/12fc37c6522e98ea8d94c5491328557ea0fc1358/conda_build/build.py#L3143
  * I think it will only be run for R recipes. I'm less certain about this, because I had trouble finding where the analogous `pl_files` and `lua_files` were defined, so this might not be true
    https://github.com/conda/conda-build/blob/12fc37c6522e98ea8d94c5491328557ea0fc1358/conda_build/post.py#L761

At minimum I know `run_test.r` will work with R recipes, since I just recently added it to r-git2r-feedstock in https://github.com/conda-forge/r-git2r-feedstock/pull/44/commits/f9d4ff15f94184b726bba6d14cb77b609a6be264

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jdblischak** [comment]
@beeankha is there any interest in reviewing this documentation-only PR? I see that you keep merging in main, but this PR doesn't touch the code, so it's not going to make a difference in the CI checks.

File: news/5479-better-document-run_test.r [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* Better document `run_test.r` (#5479)
```
Comment (beeankha) [inline news/5479-better-document-run_test.r:15]:
```suggestion
* Better document `run_test.r`. (#5479)
```

**Comment by kathatherine** [comment]
Thanks so much for helping to improve our documentation! I'll approve once the tests run green.

**Review by beeankha — APPROVED** [review]

**Comment by kathatherine** [comment]
@beeankha I know this is just a doc update and it shouldn't affect the CI tests (and also you approved it). Is that failing test a known issue?

**Comment by beeankha** [comment]
> @beeankha I know this is just a doc update and it shouldn't affect the CI tests (and also you approved it). Is that failing test a known issue?

The previously-failing tests were a known issue that have been resolved, and now we're just dealing with some CI infrastructure hiccups (i.e. network errors vs test failures). The latest re-run should hopefully come back all green.

**Review by kathatherine — APPROVED** [review]
ty!

**Pull request merged**